tests/storage/snapshot.go

```diff
@@ -991,58 +1007,41 @@ var _ = SIGDescribe("VirtualMachineSnapshot Tests", func() {
 				Expect(err).ToNot(HaveOccurred())
 				Expect(ss.Status).To(BeNil())
 
-				Eventually(func() bool {
+				Eventually(func() *snapshotv1.VirtualMachineSnapshotStatus {
 					ss, err = virtClient.VirtualMachineSnapshot(ss.Namespace).Get(context.Background(), ss.Name, metav1.GetOptions{})
 					Expect(err).ToNot(HaveOccurred())
-					if ss.Status == nil {
-						return false
-					}
-					if *ss.Status.SourceUID != *origStatus.SourceUID ||
-						*ss.Status.VirtualMachineSnapshotContentName != *origStatus.VirtualMachineSnapshotContentName ||
-						*ss.Status.CreationTime != *origStatus.CreationTime ||
-						ss.Status.Phase != origStatus.Phase ||
-						*ss.Status.ReadyToUse != *origStatus.ReadyToUse {
-						return false
-					}
-					if len(ss.Status.Conditions) != len(origStatus.Conditions) {
-						return false
-					}
-					for i, c := range ss.Status.Conditions {
-						oc := origStatus.Conditions[i]
-						if c.Type != oc.Type ||
-							c.Status != oc.Status {
-							return false
-						}
-					}
-					if len(ss.Status.Indications) != len(origStatus.Indications) {
-						return false
-					}
-					for i := range ss.Status.Indications {
-						if ss.Status.Indications[i] != origStatus.Indications[i] {
-							return false
-						}
-					}
-					return true
-				}, 180*time.Second, time.Second).Should(BeTrue())
+					return ss.Status
+				}, 180*time.Second, time.Second).Should(gstruct.PointTo(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+					"SourceUID":                         gstruct.PointTo(Equal(*origStatus.SourceUID)),
+					"VirtualMachineSnapshotContentName": gstruct.PointTo(Equal(*origStatus.VirtualMachineSnapshotContentName)),
+					"CreationTime":                      gstruct.PointTo(Equal(*origStatus.CreationTime)),
+					"Phase":                             Equal(origStatus.Phase),
+					"ReadyToUse":                        gstruct.PointTo(Equal(*origStatus.ReadyToUse)),
+					"Conditions":                        ContainElements(origStatus.Conditions),
```

Comment:
its not working with `HaveExactElements`, because as you mentioned there is the `LastTransitionTime` which is different. Not sure how else to check this..

Reply:
any suggestion @AlonaKaplan ?

Reply:
You can clear the `LastTransitionTime` from both the original and the returned status.

Reply:
Thanks it worked, I was thinking of complicated options of using the matches instead of thinking simple :)
pushed an extra commit for fixes